enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,587 @@
+
+**Integration:** Integration with the product is strengthened, meaning that adding new features becomes more straightforward. Employing tightly coupled methods could lead to poor practices.
+
+**Alerting:** The controller can notify users through various channels:
+
+- HC status.conditions
+- Log entries
+- Events
+- Alerting
+- Metrics
+
+**Validations:** In this scenario, we incorporate an additional layer of validations (CEL), managed by api-machinery. This layer promptly detects any discrepancies with the API consensus and triggers early failure. Furthermore, additional validations can be employed within the controller.
+
+**Reusability:** Leveraging the existing ETCD backup API developed by the ETCD team, we will integrate the entire API into a new structure within our ETCD API group. Within the same group, we will incorporate the necessary functionality to upload binaries to remote storage such as S3 or Azure.
+
+#### Cons
+
+**Supportability:** Development of the API and support of it during the product life-cycle.
+**Complexity:** The Hosted Control Planes team will take the responsibility to stay tuned over the API changes of the ETCD Backup.
+
+
+## Proposal
+
+The proposal commences with the acknowledgement that a user requires an ETCD backup.
+
+The primary requirement is that the ETCD `managementType` must be configured as `managed`. Once this configuration is set, further progress can proceed.
+
+This backup operation will be specified within the `hostedcluster.spec.etcd` API group under a new subgroup named `backup`. This subgroup will encompass two fields, with the primary one designated for the backup process itself, directly imported from the original ETCD Backup API. Comprehensive details regarding this API can be found in the official ETCD proposal and its corresponding implementation documentation.
+
+- [ETCD Operator Backup API Design Doc](https://docs.google.com/document/d/1xBnK3D7MCblFgxQ8RUZFyWNIU_bnSFHciLhhHAeP0Xk/edit)
+- [ETCD Backup API implementation](https://github.com/openshift/api/pull/1482)
+- [Periodic Backup Controller](https://github.com/openshift/cluster-etcd-operator/pull/1074)
+
+The other field it's called `storeEndpoint` which basically will take care of the export of the binary to an external endpoint. This endpoint could be S3, Azure, etc.... Inside of this `storeEndpoint` we have some fields:
+
+- **BackupPrefix:** Where the user could set some kind of prefix to identify that backup in a more readable way. This Prefix will be parsed, validated and used in the final name which will contain the **timestamp**, the **HostedCluster name** and the provided **BackupPrefix**.
+- **Credentials:** This is a `SecretReference` located in the HostedCluster namespace using the same name as the backup plus the "credentials" suffix
+- **ServiceConfig:** This ServiceConfig field is another struct which it will contain the details about the remote destination you will upload that backup.
+
```

Comment:
To me, it would be more useful if the new API is presented as a go struct with comments for the fields

Reply:
Not sure I'm following, could you elaborate that?

Reply:
Makes sense.